src/common.ts

```diff
@@ -11,3 +11,21 @@
 export type FilePath = string;
 /** Type alias to indicate the path of a directory. */
 export type DirPath = string;
+
+/**
+ * Use this with `.catch()` on promises where the error is already
+ * handled elsewhere, but where you would like to debug log the
+ * condition.
+ */
+export function catchFloatingPromise(err: Error) {
+  console.debug("Caught an unhandled promise " + (new Error()).stack);
```

Comment:
Should this not be logging the stack from `err`? Also, according to the docblock, it sounds like you need to throw `err` once logged, so it can be handled "elsewhere"

Reply:
OK, I have attempted to clarify the comment text.

Reply:
Ohh k. Hopefully we can refactor this out at some point and just handle the call chain more effectively, but we've got more important things to get done. :+1: